### PR TITLE
 [HDRP] Fix issue with raytracing resources not properly initialize

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
Forward port of https://github.com/Unity-Technologies/Graphics/pull/4268

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs [hunk 1]
```diff
@@ -561,23 +567,74 @@ public void OnProcessShader(Shader shader, ShaderSnippetData snippet, IList<Shad
...
+            else
+                s_ComputeShaderCache = new Dictionary<int, ComputeShader>();
+
+            if (HDRenderPipelineGlobalSettings.Ensure(canCreateNewAsset: false) == null)
+                return;
+
+            if (HDRenderPipelineGlobalSettings.instance.renderPipelineRayTracingResources == null)
+                return;
+
+            foreach (var fieldInfo in HDRenderPipelineGlobalSettings.instance.renderPipelineRayTracingResources.GetType().GetFields(BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static))
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs:615]: Same `BindingFlags.Static` not needed

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 2]
```diff
@@ -439,21 +439,15 @@ void UpgradeResourcesIfNeeded()
...
-            {
-                int qualityLevelCount = QualitySettings.names.Length;
-                for (int i = 0; i < qualityLevelCount && !requiresRayTracingResources; ++i)
-                {
-                    var hdrpAsset = QualitySettings.GetRenderPipelineAssetAt(i) as HDRenderPipelineAsset;
-                    if (hdrpAsset != null && hdrpAsset.currentPlatformRenderPipelineSettings.supportRayTracing)
-                        requiresRayTracingResources = true;
-                }
+                var hdrpAsset = QualitySettings.GetRenderPipelineAssetAt(i) as HDRenderPipelineAsset;
+                if (hdrpAsset != null && hdrpAsset.currentPlatformRenderPipelineSettings.supportRayTracing)
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:447]: same check if we need to check hardware compatibility too

**Review by RSlysz — APPROVED** [review]

**Review by jenniferd-unity — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs (hunk 2 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:447]: We must not check hardware as a build could be done on a DX12 compatible computer which don't have RTX card.

**Review by remi-chapelain — APPROVED** [review]
- Try to use wizard to make project HDRP + DXR ✔️ 
- Tried to build with raytracing ready HDRP asset by default and switched quality settings in player✔️ 
- Tried to build with non ray tracing ready HDRP asset by default and switched quality settings in player✔️

**Pull request merged**